File: Modules/ERPInterface/SIE.Web.ERPInterface/UploadTranscationRules/Commands/TransactionWarehouseLookupCommand.cs
using SIE.Domain;
using SIE.ERPInterface.Common.UploadTransactionRules;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Text;

namespace SIE.Web.ERPInterface.UploadTransactionRules.Commands
{
    /// <summary>
    /// 交易上传规则选择仓库
    /// </summary>
    public class TransactionWarehouseLookupCommand : ViewCommand
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var meta = ClientEntities.Find(args.Type);
            var savedData = RF.Find(meta.EntityType).NewList();
            var transcWarehouseList = args.Data.ToJsonObject<List<UploadTransactionExclWh>>();
            Check.NotNullOrEmpty(transcWarehouseList, nameof(transcWarehouseList));
            if (transcWarehouseList == null || transcWarehouseList.Count == 0)
            {
                throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(transcWarehouseList)));
            }
            foreach (var item in transcWarehouseList)
            {
                var transcWarehouse = new UploadTransactionExclWh();
                transcWarehouse.WarehouseId = item.WarehouseId;
                transcWarehouse.UploadTransactionRuleId = item.UploadTransactionRuleId;
                savedData.Add(transcWarehouse);
            }
            RF.Save(savedData);
            return true;
        }
    }
}

File: Modules/ERPInterface/SIE.Web.ERPInterface/UploadTranscationRules/Commands/TransactionWarehouseLookupCommand.js
SIE.defineCommand('SIE.Web.ERPInterface.UploadTransactionRules.Commands.TransactionWarehouseLookupCommand', {
    extend: 'SIE.cmd.LookupCommandBase',
    userConfig: {
        dataParams: { specKeyPrototyName: 'WarehouseId', targetClassName: 'SIE.Warehouses.Warehouse' },
    },
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    save: function (win) {
        var me = this;
        var indata = {};
        var selections = this._targetView.getSelection();
        if (selections!=null && selections.length > 0) {
            var operationDatas = [];
            SIE.each(selections, function (item) {
                var warehouseId = item.getId();
                if (me._sourceViewSelectItems.indexOf(warehouseId) === -1) {
                    var transcWarehouse = { UploadTransactionRuleId: me._sourceId, WarehouseId: warehouseId };
                    operationDatas.push(transcWarehouse);
                }
            });
            indata = operationDatas;
            me._targetView.execute({
                data: indata,
                success: function (res) {
                    win.close();  //关闭模态窗口
                    me._ownerView.loadChildData(true); //重载视图数据
                }
            }, me._ownerView);
        }
        else {
            SIE.Msg.showWarning('没有可提交的数据'.t());
        }
    }
    // end 
});